src/Language/PureScript/Bridge/Printer.hs

```diff
@@ -83,9 +83,9 @@ sumTypeToText st@(SumType t cs) = T.unlines $
     genericConstrainsInner = T.intercalate ", " $ map genericInstance sumTypeParameters
     stpLength = length sumTypeParameters
     bracketWrap x = "(" <> x <> ")"
-    sumTypeParameters = filter isTypeParameter $ getUsedTypes st
-    -- TODO: I find this stupid. Would much rather like some other way to make this check?
-    isTypeParameter tp = T.toLower (_typeName tp) == _typeName tp
+    sumTypeParameters = filter isTypeParam . Set.toList $ getUsedTypes st
+    -- NOTE: only this captures monadic type parameters. See test/Spec.hs
+    isTypeParam typ = T.toLower (_typeName typ) == _typeName typ
```

Comment:
You are right! But it should work if you don't compare the full TypeInfo's but only their `_typeName`. This would cover your case. The used types contain a TypeInfo named m (with type parameter b - which is fine) and b itself - both would be found and contained in the constraints.